### PR TITLE
Deep improvement of WS api/issues

### PR DESCRIPTION
**Pull request by simonbrandhof**
The target is to bring issue multiple locations (SONAR-6052). To achieve that,
some intermediate steps are required :

- do not use Elasticsearch to load issue data. Just to search for 
   the matching issue keys. Data must be loaded from DB.
   That allows to not store structured advanced locations in ES.
- refactor some ruby WS to Java in order to improve the way 
   to format WS responses
- use protobuf specification in order to define and write WS response
- SONAR-6582 sanitize responses of api/issues/{operation}
- SONAR-6717 stop support of plugin actions

### CODE REVIEW

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+    private int endOffset_;
+    /**
+     * <code>optional int32 end_offset = 4;</code>
+     *
+     * <pre>
+     * If absent it means range ends at the last offset of end line
+     * </pre>
+     */
+    public boolean hasEndOffset() {
+      return ((bitField0_ & 0x00000008) == 0x00000008);
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+    public static final int END_OFFSET_FIELD_NUMBER = 4;
+    private int endOffset_;
+    /**
+     * <code>optional int32 end_offset = 4;</code>
+     *
+     * <pre>
+     * If absent it means range ends at the last offset of end line
+     * </pre>
+     */
+    public boolean hasEndOffset() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 3]
```diff
@@ -0,0 +1,818 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
+// source: db-commons.proto
+
+package org.sonar.db.protobuf;
+
+public final class DbCommons {
+  private DbCommons() {}
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Move this closing curly brace to the next line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+  public static void registerAllExtensions(
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1186)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 5]
```diff
@@ -0,0 +1,818 @@
...
+        }
+        this.mergeUnknownFields(other.getUnknownFields());
+        return this;
+      }
+
+      public final boolean isInitialized() {
+        return true;
+      }
+
+      public Builder mergeFrom(
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 6]
```diff
@@ -0,0 +1,818 @@
...
+      return ((bitField0_ & 0x00000008) == 0x00000008);
+    }
+    /**
+     * <code>optional int32 end_offset = 4;</code>
+     *
+     * <pre>
+     * If absent it means range ends at the last offset of end line
+     * </pre>
+     */
+    public int getEndOffset() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 7]
```diff
@@ -0,0 +1,818 @@
...
+          setStartOffset(other.getStartOffset());
+        }
+        if (other.hasEndOffset()) {
+          setEndOffset(other.getEndOffset());
+        }
+        this.mergeUnknownFields(other.getUnknownFields());
+        return this;
+      }
+
+      public final boolean isInitialized() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MINOR "final" is redundant in this context. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2333)
MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 8]
```diff
@@ -0,0 +1,818 @@
...
+        } finally {
+          if (parsedMessage != null) {
+            mergeFrom(parsedMessage);
+          }
+        }
+        return this;
+      }
+      private int bitField0_;
+
+      private int startLine_ ;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Rename this field "startLine_" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 9]
```diff
@@ -0,0 +1,818 @@
...
+      return ((bitField0_ & 0x00000004) == 0x00000004);
+    }
+    /**
+     * <code>optional int32 start_offset = 3;</code>
+     *
+     * <pre>
+     * If absent it means range starts at the first offset of start line
+     * </pre>
+     */
+    public int getStartOffset() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 10]
```diff
@@ -0,0 +1,818 @@
...
+          parsedMessage = (org.sonar.db.protobuf.DbCommons.TextRange) e.getUnfinishedMessage();
+          throw e;
+        } finally {
+          if (parsedMessage != null) {
+            mergeFrom(parsedMessage);
+          }
+        }
+        return this;
+      }
+      private int bitField0_;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Rename this field "bitField0_" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 11]
```diff
@@ -0,0 +1,818 @@
...
+     * <pre>
+     * If absent it means range starts at the first offset of start line
+     * </pre>
+     */
+    public int getStartOffset() {
+      return startOffset_;
+    }
+
+    public static final int END_OFFSET_FIELD_NUMBER = 4;
+    private int endOffset_;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Rename this field "endOffset_" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 12]
```diff
@@ -0,0 +1,818 @@
...
+        result.startLine_ = startLine_;
+        if (((from_bitField0_ & 0x00000002) == 0x00000002)) {
+          to_bitField0_ |= 0x00000002;
+        }
+        result.endLine_ = endLine_;
+        if (((from_bitField0_ & 0x00000004) == 0x00000004)) {
+          to_bitField0_ |= 0x00000004;
+        }
+        result.startOffset_ = startOffset_;
+        if (((from_bitField0_ & 0x00000008) == 0x00000008)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 13]
```diff
@@ -0,0 +1,818 @@
...
+    public final boolean isInitialized() {
+      byte isInitialized = memoizedIsInitialized;
+      if (isInitialized == 1) return true;
+      if (isInitialized == 0) return false;
+
+      memoizedIsInitialized = 1;
+      return true;
+    }
+
+    public void writeTo(com.google.protobuf.CodedOutputStream output)
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 14]
```diff
@@ -0,0 +1,818 @@
...
+        int to_bitField0_ = 0;
+        if (((from_bitField0_ & 0x00000001) == 0x00000001)) {
+          to_bitField0_ |= 0x00000001;
+        }
+        result.startLine_ = startLine_;
+        if (((from_bitField0_ & 0x00000002) == 0x00000002)) {
+          to_bitField0_ |= 0x00000002;
+        }
+        result.endLine_ = endLine_;
+        if (((from_bitField0_ & 0x00000004) == 0x00000004)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 15]
```diff
@@ -0,0 +1,818 @@
...
+      if (isInitialized == 0) return false;
+
+      memoizedIsInitialized = 1;
+      return true;
+    }
+
+    public void writeTo(com.google.protobuf.CodedOutputStream output)
+                        throws java.io.IOException {
+      getSerializedSize();
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 16]
```diff
@@ -0,0 +1,818 @@
...
+        output.writeInt32(1, startLine_);
+      }
+      if (((bitField0_ & 0x00000002) == 0x00000002)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 17]
```diff
@@ -0,0 +1,818 @@
...
+  static {
+    java.lang.String[] descriptorData = {
+      "\n\020db-commons.proto\022\024sonarqube.db.commons" +
+      "\"[\n\tTextRange\022\022\n\nstart_line\030\001 \001(\005\022\020\n\010end" +
+      "_line\030\002 \001(\005\022\024\n\014start_offset\030\003 \001(\005\022\022\n\nend" +
+      "_offset\030\004 \001(\005B\031\n\025org.sonar.db.protobufH\001"
+    };
+    com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner assigner =
+        new com.google.protobuf.Descriptors.FileDescriptor.    InternalDescriptorAssigner() {
+          public com.google.protobuf.ExtensionRegistry assignDescriptors(
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 18]
```diff
@@ -0,0 +1,818 @@
...
+      descriptor;
+  static {
+    java.lang.String[] descriptorData = {
+      "\n\020db-commons.proto\022\024sonarqube.db.commons" +
+      "\"[\n\tTextRange\022\022\n\nstart_line\030\001 \001(\005\022\020\n\010end" +
+      "_line\030\002 \001(\005\022\024\n\014start_offset\030\003 \001(\005\022\022\n\nend" +
+      "_offset\030\004 \001(\005B\031\n\025org.sonar.db.protobufH\001"
+    };
+    com.google.protobuf.Descriptors.FileDescriptor.InternalDescriptorAssigner assigner =
+        new com.google.protobuf.Descriptors.FileDescriptor.    InternalDescriptorAssigner() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Make this a named "static" inner class. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2694)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 19]
```diff
@@ -0,0 +1,818 @@
...
+        if (((from_bitField0_ & 0x00000008) == 0x00000008)) {
+          to_bitField0_ |= 0x00000008;
+        }
+        result.endOffset_ = endOffset_;
+        result.bitField0_ = to_bitField0_;
+        onBuilt();
+        return result;
+      }
+
+      public Builder mergeFrom(com.google.protobuf.Message other) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 20]
```diff
@@ -0,0 +1,818 @@
...
+    public void writeTo(com.google.protobuf.CodedOutputStream output)
+                        throws java.io.IOException {
+      getSerializedSize();
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
+        output.writeInt32(1, startLine_);
+      }
+      if (((bitField0_ & 0x00000002) == 0x00000002)) {
+        output.writeInt32(2, endLine_);
+      }
+      if (((bitField0_ & 0x00000004) == 0x00000004)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 21]
```diff
@@ -0,0 +1,818 @@
...
+        output.writeInt32(3, startOffset_);
+      }
+      if (((bitField0_ & 0x00000008) == 0x00000008)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 22]
```diff
@@ -0,0 +1,818 @@
...
+        if (other instanceof org.sonar.db.protobuf.DbCommons.TextRange) {
+          return mergeFrom((org.sonar.db.protobuf.DbCommons.TextRange)other);
+        } else {
+          super.mergeFrom(other);
+          return this;
+        }
+      }
+
+      public Builder mergeFrom(org.sonar.db.protobuf.DbCommons.TextRange other) {
+        if (other == org.sonar.db.protobuf.DbCommons.TextRange.getDefaultInstance()) return this;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MINOR Missing curly brace. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00121)
MAJOR Change this comparison to use the equals method. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1698)
MAJOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 23]
```diff
@@ -0,0 +1,818 @@
...
+    }
+
+    // @@protoc_insertion_point(class_scope:sonarqube.db.commons.TextRange)
+  }
+
+  private static final com.google.protobuf.Descriptors.Descriptor
+    internal_static_sonarqube_db_commons_TextRange_descriptor;
+  private static
+    com.google.protobuf.GeneratedMessage.FieldAccessorTable
+      internal_static_sonarqube_db_commons_TextRange_fieldAccessorTable;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MINOR Rename this field "internal_static_sonarqube_db_commons_TextRange_fieldAccessorTable" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS3008)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 24]
```diff
@@ -0,0 +1,818 @@
...
+    }
+
+    private void initFields() {
+      startLine_ = 0;
+      endLine_ = 0;
+      startOffset_ = 0;
+      endOffset_ = 0;
+    }
+    private byte memoizedIsInitialized = -1;
+    public final boolean isInitialized() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)
MINOR "final" is redundant in this context. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2333)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 25]
```diff
@@ -0,0 +1,818 @@
...
+      byte isInitialized = memoizedIsInitialized;
+      if (isInitialized == 1) return true;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MINOR Missing curly brace. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00121)
MAJOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 26]
```diff
@@ -0,0 +1,818 @@
...
+      if (isInitialized == 0) return false;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)
MINOR Missing curly brace. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00121)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 27]
```diff
@@ -0,0 +1,818 @@
...
+      }
+      if (((bitField0_ & 0x00000002) == 0x00000002)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeInt32Size(2, endLine_);
+      }
+      if (((bitField0_ & 0x00000004) == 0x00000004)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeInt32Size(3, startOffset_);
+      }
+      if (((bitField0_ & 0x00000008) == 0x00000008)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 28]
```diff
@@ -0,0 +1,818 @@
...
+      size = 0;
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeInt32Size(1, startLine_);
+      }
+      if (((bitField0_ & 0x00000002) == 0x00000002)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeInt32Size(2, endLine_);
+      }
+      if (((bitField0_ & 0x00000004) == 0x00000004)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 29]
```diff
@@ -0,0 +1,818 @@
...
+      }
+      /**
+       * <code>optional int32 start_line = 1;</code>
+       *
+       * <pre>
+       * Start line. Should never be absent
+       * </pre>
+       */
+      public Builder clearStartLine() {
+        bitField0_ = (bitField0_ & ~0x00000001);
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 30]
```diff
@@ -0,0 +1,818 @@
...
+        startLine_ = 0;
+        onChanged();
+        return this;
+      }
+
+      private int endLine_ ;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Rename this field "endLine_" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00116)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 31]
```diff
@@ -0,0 +1,818 @@
...
+        output.writeInt32(3, startOffset_);
+      }
+      if (((bitField0_ & 0x00000008) == 0x00000008)) {
+        output.writeInt32(4, endOffset_);
+      }
+      getUnknownFields().writeTo(output);
+    }
+
+    private int memoizedSerializedSize = -1;
+    public int getSerializedSize() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 32]
```diff
@@ -0,0 +1,818 @@
...
+
+      private int endLine_ ;
+      /**
+       * <code>optional int32 end_line = 2;</code>
+       *
+       * <pre>
+       * End line (inclusive). Absent means it is same as start line
+       * </pre>
+       */
+      public boolean hasEndLine() {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Add the "@Override" annotation above this method signature [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1161)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 33]
```diff
@@ -0,0 +1,818 @@
...
+    public int getSerializedSize() {
+      int size = memoizedSerializedSize;
+      if (size != -1) return size;
+
+      size = 0;
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeInt32Size(1, startLine_);
+      }
+      if (((bitField0_ & 0x00000002) == 0x00000002)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 34]
```diff
@@ -0,0 +1,818 @@
...
+      private int endLine_ ;
+      /**
+       * <code>optional int32 end_line = 2;</code>
+       *
+       * <pre>
+       * End line (inclusive). Absent means it is same as start line
+       * </pre>
+       */
+      public boolean hasEndLine() {
+        return ((bitField0_ & 0x00000002) == 0x00000002);
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 35]
```diff
@@ -0,0 +1,818 @@
...
+      getUnknownFields().writeTo(output);
+    }
+
+    private int memoizedSerializedSize = -1;
+    public int getSerializedSize() {
+      int size = memoizedSerializedSize;
+      if (size != -1) return size;
+
+      size = 0;
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MAJOR Remove those useless parentheses. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AUselessParenthesesCheck)

File: sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java [hunk 36]
```diff
@@ -0,0 +1,818 @@
...
+      if (((bitField0_ & 0x00000008) == 0x00000008)) {
+        output.writeInt32(4, endOffset_);
+      }
+      getUnknownFields().writeTo(output);
+    }
+
+    private int memoizedSerializedSize = -1;
+    public int getSerializedSize() {
+      int size = memoizedSerializedSize;
+      if (size != -1) return size;
```
Comment (SonarTech) [inline sonar-db/src/main/gen-java/org/sonar/db/protobuf/DbCommons.java]: MINOR Missing curly brace. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00121)
MAJOR At most one statement is allowed per line, but 2 statements were found on this line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00122)

**Comment by teryk** [comment]
Can't LGTM but it seems good. When SQ build and CI pass and issues are fixed I suggest to put it in master. It must be thoroughly tested by our PO (issues/search but also the WS migrated to java).

**Pull request closed without merging**